include/bl_storage_internal.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef BL_STORAGE_INTERNAL_H_
+#define BL_STORAGE_INTERNAL_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @defgroup bl_storage Bootloader storage (protected data).
+ * @{
+ */
+
+const struct monotonic_counter *get_counter_struct(uint16_t description);
+
+/** This library implements monotonic counters where each time the counter
+ *  is increased, a new slot is written.
+ *  This way, the counter can be updated without erase. This is, among other things,
+ *  necessary so the counter can be used in the OTP section of the UICR
+ *  (available on e.g. nRF91 and nRF53).
+ */
+struct monotonic_counter {
+	/* Counter description. What the counter is used for. See
+	 * BL_MONOTONIC_COUNTERS_DESC_x.
+	 */
+	uint32_t description;
+	/* Number of entries in 'counter_slots' list. */
+	uint32_t num_counter_slots;
+#ifdef CONFIG_NRFX_NVMC
+	uint16_t counter_slots[1];
+#elif defined(CONFIG_NRFX_RRAMC)
+	uint32_t counter_slots[1];
+#endif
+};
+
+/** The second data structure in the provision page. It has unknown length since
+ *  'counters' is repeated. Note that each entry in counters also has unknown
+ *  length, and each entry can have different length from the others, so the
+ *  entries beyond the first cannot be accessed via array indices.
+ */
+struct counter_collection {
+	uint32_t type; /* Must be "monotonic counter". */
+	uint32_t num_counters; /* Number of entries in 'counters' list. */
```

Comment:
Same here: These have been changed from uint16_t to uin32_t for all devices, so all APIs should be changed as well IMO, as well as changing reading from halfword_read to halfword_write.

Reply:
Will change these back